lesson_13/homework_13.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, List, Dict
+import csv
+import json
+import logging
+from pathlib import Path
+import xml.etree.ElementTree as ET
+
+def setup_logger(log_filename: str, level=logging.INFO) -> logging.Logger:
+    logger = logging.getLogger(log_filename)
+    logger.setLevel(level)
+    file_handler = logging.FileHandler(log_filename, mode='a', encoding='utf-8')
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+    return logger
+
+
+def read_file(filepath: Path, file_type: str) -> Any:
+    if not filepath.exists():
+        print(f"File {filepath} not found!")
+        return None
+    
+    if file_type == 'csv':
+        with open(filepath, "r", encoding="utf8") as file:
+            reader = csv.reader(file, delimiter=";")
+            return list(reader)
+    
+    elif file_type == 'json':
+        with open(filepath, "r", encoding="utf8") as file:
+            try:
+                return json.load(file)
+            except json.decoder.JSONDecodeError:
+                return None
+    
+    elif file_type == 'xml':
+        try:
+            tree = ET.parse(filepath)
+            return tree.getroot()
+        except ET.ParseError:
+            return None
+
+def write_file(filepath: Path, content: Any, file_type: str):
+    if content is None:
+        print(f"No content to write to {filepath}")
+        return
+    
+    if file_type == 'csv' and isinstance(content, list):
+        with open(filepath, "w", encoding="utf8", newline="") as file:
+            writer = csv.writer(file, delimiter=";")
+            writer.writerows(content)
+    
+    elif file_type == 'json' and isinstance(content, dict):
```

Comment:
```suggestion
    if filepath.suffix == 'csv' and isinstance(content, list):
        with open(filepath, "w", encoding="utf8", newline="") as file:
            writer = csv.writer(file, delimiter=";")
            writer.writerows(content)
    
    elif filepath.suffix == 'json' and isinstance(content, dict):
```
можна зразу дивитися на розширення ф-лу і зменшити к-сть параметрів